### ISSUE TITLE
Contribute to PyTorch?

### ISSUE THREAD

**Issue by adamjstewart** [RElbers/info-nce-pytorch#10]
Have you ever considered contributing this loss function to [PyTorch](https://pytorch.org/docs/stable/nn.html#loss-functions)? TorchGeo would love to use it for SimCLR/MoCo SSL but we are trying to avoid adding yet another dependency. If it was built into PyTorch, it would be much easier to use.

**Comment by RElbers** [comment]
I understand the desire to limit the number of dependencies. I'll see what I can do.

**Comment by Yuntian9708** [comment]
I have a question about codes for negative key =None:
 # If negative key is None, then Negative keys are implicitly off-diagonal positive keys.

        # Cosine between all combinations
        logits = query @ transpose(positive_key)

        # Positive keys are the entries on the diagonal
        labels = torch.arange(len(query), device=query.device)

why labels are torch.arrange(len(query)? for example: 0,1,2,3,4,5...I think labels for query-postive key should be torch.ones(len(query))

**Comment by adamjstewart** [comment]
We ended up using the [lightly](https://docs.lightly.ai/self-supervised-learning/lightly.loss.html#lightly.loss.ntx_ent_loss.NTXentLoss) implementation because we needed it anyway for other SSL features. I'll close this since I personally no longer have a need for this, but I still think it would be cool if this loss was in PyTorch someday.
